### PR TITLE
[SIMULATION] [CLANG] Fix warnings reported by llvm16 in CLANG IBs

### PR DESCRIPTION
**Pull request by aandvalenzuela**
Hello,

I am opening PRs for simulation modules again (like in https://github.com/cms-sw/cmssw/pull/41584) since we already checked from where the difference in comparison were coming from at https://github.com/cms-sw/cmssw/pull/41653.

Thanks,
Andrea.

### CODE REVIEW

File: SimG4CMS/HcalTestBeam/plugins/HcalTB04Analysis.cc [hunk 1]
```diff
@@ -974,32 +978,40 @@ void HcalTB04Analysis::fillEvent(PHcalTB04Info& product) {
   product.setLongProf(eslay, eqlay);
 
   //Save Hits
-  int i, nhit = 0;
+  int nhit = 0;
+#ifdef EDM_ML_DEBUG
+  i = 0;
```
Comment (smuzaffar) [inline SimG4CMS/HcalTestBeam/plugins/HcalTB04Analysis.cc:983]: @aandvalenzuela , shouldn't this be `int i=0;` ?

Comment (aandvalenzuela) [reply SimG4CMS/HcalTestBeam/plugins/HcalTB04Analysis.cc:983]: Ah nono, I am wrong sorry. I am changing this :)

**Comment by aandvalenzuela** [comment]
please test

**Comment by bsunanda** [comment]
Please check #41671. It is correctly done there
________________________________

**Pull request closed without merging**